sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/pixels/SyncDailyReportingWorker.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.impl.pixels
+
+import android.content.Context
+import androidx.lifecycle.LifecycleOwner
+import androidx.work.BackoffPolicy
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.duckduckgo.anvil.annotations.ContributesWorker
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.app.lifecycle.MainProcessLifecycleObserver
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesMultibinding
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeUnit.HOURS
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+@ContributesWorker(AppScope::class)
+class SyncDailyReportingWorker(
+    context: Context,
+    workerParameters: WorkerParameters,
+) : CoroutineWorker(context, workerParameters) {
+
+    @Inject
+    lateinit var syncPixels: SyncPixels
+
+    @Inject
+    lateinit var dispatchers: DispatcherProvider
+
+    override suspend fun doWork(): Result {
+        return withContext(dispatchers.io()) {
+            syncPixels.fireStatsPixel()
+            return@withContext Result.success()
+        }
+    }
+}
+
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = MainProcessLifecycleObserver::class,
+)
+class SyncDailyReportingWorkerScheduler @Inject constructor(
+    private val workManager: WorkManager,
+) : MainProcessLifecycleObserver {
+
+    override fun onCreate(owner: LifecycleOwner) {
+        super.onCreate(owner)
+        Timber.v("Scheduling sync daily reporting worker")
+        val workerRequest = PeriodicWorkRequestBuilder<SyncDailyReportingWorker>(24, HOURS)
```

Comment:
there's a question about scheduling every 24h here and then filtering by `today()`.

Reply:
answered above